lib/msf/core/post/linux/system.rb

```diff
@@ -96,6 +96,33 @@ def get_sysinfo
     return system_data
   end
 
+  #
+  # return true if possible chroot environment detect, if not true then false
+  #
+  def is_chroot?
+    chroot_env = false
+
+    if not directory?("/proc/")
+      chroot_env = true
+    else
+      dir("/proc/").each do |pid|
+        pid = Integer(pid) rescue nil
+        if pid
+          filename = ::File.readlink("/proc/#{pid}/exe") rescue nil
```

Comment:
oh, sorry, I search here for a readlink function in metasploit API, thats work on client side, but I found nothing, i think open the files (/proc/[pid]/exe), to check if exists, if really dont have a readlink function, of course